flow/ProtocolVersions.cmake

```diff
@@ -8,11 +8,20 @@
 # used and should not be changed from 0.
 #                                                          xyzdev
 #                                                          vvvv
+
+if (COMPATIBLE_WITH_SNOWFLAKE_RELEASES)
+set(FDB_PV_DEFAULT_VERSION                      "0x0FDB00C072000000LL")
```

Comment:
This creates inconsistency between protocol version numbering and build version numbering, creating potentially further confusion.

We should switch the build versions as well if the flag is on.

Reply:
It does not. It consistently follows the open-source versioning scheme, just changing the flag to make it pass the version check. If we want to make a full-fledged implementation following Snowflakes versioning scheme, the change will be much larger. We also need to adjust then the feature table below, also API Version and API feature table.

Reply:
The inconsistency I'm pointing to is that today, if you have a binary that reports 0x0FDB00C.. as a protocol version, its build version is 71.X. 

With this change, you can have a binary with 0x0FDB00C... protocol version which will have 71.X version if built from a release branch, but will have a version 7.X if built from main. This is inconsistent.

What would be consistent is that a binary with 0x0FDB00C... protocol version has 71.X version regardless of whether it's a release or pre-release (built from main). 

As much as I dislike the Snowflake release scheme, I would much prefer we don't introduce new inconsistencies in produced artifacts.

How much larger a change would it be to use Snowflake build versioning scheme in main under this flag?

Reply:
This what we have done for the version change:
https://github.com/apple/foundationdb/pull/8028/
https://github.com/apple/foundationdb/pull/8163/

Of course, part of these changes is refactoring to make future version changes easier